py/client-ticking/build.gradle

```diff
@@ -0,0 +1,114 @@
+plugins {
+    id 'com.bmuschko.docker-remote-api'
+    id 'io.deephaven.project.register'
+    id 'io.deephaven.deephaven-in-docker'
+}
+
+configurations {
+    pythonWheel
+}
+
+dependencies {
+    pythonWheel project(':py-client')
+}
+
+evaluationDependsOn(':cpp-client')
+
+def prefix = '/opt/deephaven'
+
+// start a grpc-api server
+String randomSuffix = UUID.randomUUID().toString();
+deephavenDocker {
+    envVars.set([
+        'START_OPTS':'-Xmx512m -DAuthHandlers=io.deephaven.auth.AnonymousAuthenticationHandler'
+    ])
+    containerName.set "pydeephaven-test-container-${randomSuffix}"
+    networkName.set "pydeephaven-network-${randomSuffix}"
+}
+
+def buildPyClientTicking = Docker.registerDockerTask(project, 'pyClientTicking') {
+    // Only tested on x86-64, and we only build dependencies for x86-64
+    platform = 'linux/amd64'
+
+    copyIn {
+        from(layout.projectDirectory) {
+            include 'setup.py'
+            include 'README.md'
+            include 'src/**'
+        }
+        from(configurations.pythonWheel) {
+            into 'wheels'
+        }
+    }
+    copyOut {
+        into layout.buildDirectory.dir('wheel')
+    }
+    dockerfile {
+        from('deephaven/cpp-client:local-build')
+        runCommand("""mkdir -p \\
+                        /out \\
+                        ${prefix}/log \\
+                        ${prefix}/src/py-client-ticking/src \\
+                        ${prefix}/src/py-client-ticking/in-wheels
+
+                   """)
+        copyFile('setup.py', "${prefix}/src/py-client-ticking")
+        copyFile('README.md', "${prefix}/src/py-client-ticking")
+        copyFile('src/', "${prefix}/src/py-client-ticking/src/")
+        copyFile('wheels/', "${prefix}/src/py-client-ticking/in-wheels")
```

Comment:
Any reason we change the dest dir to `in-wheels` instead of just keeping the same name `wheels`?

At least we should have the trailing slash in the destination for consistency.

Reply:
The `in-wheels` was because in that image creation script, the wheel is not one that is being created in the image and for output, but one that is coming from a different gradle target as input (to be installed in the image, as opposed to be created by it)


It is a temporary directory in the image that gets removed before the image is published; just a staging area for the wheel coming form py/client.